src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/DocumentsController.kt

```diff
@@ -31,24 +35,51 @@ class DocumentsController(
       is AuthorisableActionResult.Success -> Unit
     }
 
-    val documentsMetaDataResult = offenderService.getDocuments(crn)
-    val documentsMetaData = when (documentsMetaDataResult) {
+    val documentsFromApDeliusFlag = featureFlagService.getBooleanFlag("get-documents-from-ap-delius")
+    val documentsMetaData = when (val documentsMetaDataResult = getDocuments(retrieveDocumentsFromApDelius = documentsFromApDeliusFlag, crn)) {
       is AuthorisableActionResult.NotFound -> throw NotFoundProblem(crn, "Documents")
       is AuthorisableActionResult.Unauthorised -> throw ForbiddenProblem()
       is AuthorisableActionResult.Success -> documentsMetaDataResult.entity
     }
 
-    val documentMetaData = documentsMetaData.findDocument(documentId.toString())
-      ?: throw NotFoundProblem(documentId, "Document")
-
+    val documentFilename = getDocumentFileName(documentsMetaData, documentId)
     return ResponseEntity(
       StreamingResponseBody { outputStream ->
-        offenderService.getDocument(crn, documentId.toString(), outputStream)
+        offenderService.getDocument(
+          crn,
+          documentId.toString(),
+          outputStream,
+          isRetrieveDocumentFromApDelius = documentsFromApDeliusFlag,
```

Comment:
Tiny thing, but maybe `isDocumentFromAPDelius`? May read slightly better

Reply:
will do - thanks @Rosie-Brigham

Reply:
done and pushed